File: tests/fixtures/js_repo/src/greeter.ts
import { formatName } from "./helper";
import * as path from "path";

export class Greeter {
  greet(name: string): string {
    const cleaned = this.clean(name);
    return path.join(cleaned, "greeting");
  }

  clean(name: string): string {
    return formatName(name);
  }
}

File: tests/fixtures/java_repo/src/com/example/Greeter.java
package com.example;

import com.example.util.Formatter;
import java.util.Objects;

public class Greeter {
    public String greet(String name) {
        String cleaned = this.clean(name);
        Formatter formatter = new Formatter();
        return formatter.format(cleaned);
    }

    private String clean(String name) {
        return Objects.requireNonNull(name);
    }
}
